src/python/dxpy/scripts/dx.py

```diff
@@ -2613,6 +2613,129 @@ def handle_arg_conflicts(args):
         print("Error: %s" % (e.message,), file=sys.stderr)
         err_exit()
 
+
+# Find out the project and folder where to place compilation outputs.
+#
+# There are several legal syntaxes for destination:
+#    project-id:/folder
+#    project-id:
+#    :/folder
+#    /folder
+def compile_destination(args):
+    current_dir = get_pwd().split(':')
+    project = current_dir[0]
+    folder = current_dir[1]
+    d = args.destination
+
+    if (d is not None and
+       d != '.'):
+        if ':' in d:
+            vec = d.split(':')
+            if vec.length == 1:
+                if d.endswith(':'):
+                    project = vec[0]
+                else:
+                    folder = vec[0]
+            if vec.length == 2:
+                project = vec[0]
+                folder = vec[1]
+            if vec.length >= 3:
+                raise DXParserError('Error: invalid path {}'.format(d))
+        else:
+            folder = d
+    if folder == "":
+        folder = '/'
+    if not folder.startswith('/'):
+        raise DXParserError('Error: folder must start with slash {}'.format(d))
+    return project + ":" + folder
+
+
+# Convert each unicode character to four hexadecimal digits
+def _unicodeToHex(buf):
```

Comment:
Would be nice to know if there is a simpler way to handle unicode passed through to subprocess and then eventually execv and then in this case even into the JVM.   Seems like a lot of issues could happen along the way, so the hex encoding makes sense for now since you have control over both encoding and decoding across languages.  May want to add a comment to explain the reasoning here.